modem: cmux: replace `crc8` with `crc8_rohc` for FCS calculation

ndrs-pst opened the pull request:
This PR contains three commits as follows:
- lib: crc: add `crc8_rohc` for CRC-8/ROHC variant calculation
- tests: unit: crc: use `zassert_equal` for cleaner assertions in CRC8-CCITT and CRC8-ROHC test cases
- modem: cmux: replace `crc8` with `crc8_rohc` for FCS calculation

ndrs-pst committed:
lib: crc: add `crc8_rohc` for CRC-8/ROHC variant calculation

This commit introduces the `crc8_rohc` function to the CRC library,
implementing the CRC-8/ROHC (RObust Header Compression) variant.
This algorithm is widely used in networking protocols, which is commonly
found in modem subsystems.

Signed-off-by: Pisit Sawangvonganan <pisit@ndrsolution.com>

ndrs-pst committed:
modem: cmux: replace `crc8` with `crc8_rohc` for FCS calculation

To improve performance in FCS calculation, this commit replaces
the usage of the generic `crc8` function with the specific `crc8_rohc`
function in `modem_cmux.c`.
The `crc8_rohc` function utilizes a small table approach, enhancing
the efficiency of CRC-8/ROHC variant calculations while optimizing
memory usage.

Signed-off-by: Pisit Sawangvonganan <pisit@ndrsolution.com>

bjarki-andreasen commented on tests/unit/crc/main.c:
```diff
@@ -187,6 +187,29 @@ ZTEST(crc, test_crc8_ccitt)
+	uint8_t test4[] = { 0x07, 0x3F, 0x89 };  /* GSM 07.10 example */
+	uint8_t test5[] = { 0x03, 0x3f, 0x01, 0x1c };  /* Our GSM 07.10 calc */
+
+	zassert(crc8_rohc(CRC8_ROHC_INITIAL_VALUE, test0,
+			  sizeof(test0)) == 0xcf, "pass", "fail");
```
Please use zassert_equal() and ommit the "pass" and "fail" :) Those are implied, and zasser_equal already expands to a descriptive error by default :)

ndrs-pst replied:
Got it, I may wait for all CI passed 😃.
Do I need to correct this one ZTEST(crc, test_crc8_ccitt) as well?

ndrs-pst replied:
Already address it, also with ZTEST(crc, test_crc8_ccitt) as well.

bjarki-andreasen reviewed (dismissed):
Really cool :)

ndrs-pst committed:
tests: unit: crc: use `zassert_equal` for cleaner assertions

Replace `zassert` with `zassert_equal` in CRC8-CCITT and CRC8-ROHC
test cases, as `zassert_equal` is more appropriate for comparing
actual and expected values.

Signed-off-by: Pisit Sawangvonganan <pisit@ndrsolution.com>

bjarki-andreasen reviewed (approved):
Nice work!
